pkg/cableengine/healthchecker/pinger.go

```diff
@@ -0,0 +1,138 @@
+package healthchecker
+
+import (
+	"math"
+	"time"
+
+	"github.com/go-ping/ping"
+	"k8s.io/klog"
+)
+
+var waitTime time.Duration = 15
+var size uint64 = 1000
+
+type Pinger struct {
+	remoteEndpointName string
+	healthCheckIp      string
+	statistics         Statistics
+	Status             string
+	stopCh             chan struct{}
+	pinger             *ping.Pinger
+}
+
+type Statistics struct {
+	previousRtts []uint64
+	sum          uint64
+	mean         uint64
+	stdDev       uint64
+	lastRtt      uint64
+	minRtt       uint64
+	maxRtt       uint64
+	sqrDiff      uint64
+	index        uint64
+	size         uint64
+}
+
+func NewPinger(localEndpointName, healthCheckIP string) *Pinger {
+	return &Pinger{
+		remoteEndpointName: localEndpointName,
+		healthCheckIp:      healthCheckIP,
+		statistics: Statistics{
+			size:         size,
+			previousRtts: make([]uint64, size),
+		},
+		stopCh: make(chan struct{}),
+	}
+}
+
+func (s *Statistics) updateStatistics(rtt uint64) {
```

Comment:
I'd move all "Statistics" to statistics.go

Reply:
Done.